test/e2e/tests/test_scenarios.py

```diff
@@ -37,6 +37,39 @@ class ParameterGroupHelper(helper.ResourceHelper):
     Overrides methods as required for custom resources.
     """
 
+    def assert_items_aws(self, expectations: dict, actual: dict) -> None:
+        if not expectations:
+            # nothing to assert as there are no expectations
+            return
+        if not actual:
+            # there are expectations but no given actual state to validate
+            # following assert will fail and assert introspection will provide useful information for debugging
+            assert expectations == actual
```

Comment:
As per my previous, very long, answer - https://github.com/aws-controllers-k8s/memorydb-controller/pull/65#discussion_r1116301917

I don't believe that this should be completely overriding `assert_items_aws`. I think it's a bad OOP pattern, when we could be using smaller, more well defined, methods.

Reply:
Yeah added `assert_extra_items_aws` and `get_aws_resource` inside of `assert_items_aws`. Only override those two functions.